update mapping downloader with change in FTP mapping dir format

vansika opened the pull request:
As discussed on IRC.

vansika commented:
I think we don't need this [downloader](https://github.com/metabrainz/listenbrainz-server/blob/master/spark_manage.py#L80) func anymore because ideally request to download and upload mapping should be sent from lemmy in light of the workflow we have recently adopted. I want to open a separate PR to move all such functions to [request_manage.py](https://github.com/metabrainz/listenbrainz-server/blob/master/listenbrainz/spark/request_manage.py)

vansika commented:
Have the field names/col names/attributes in updated mapping-matchable changed?
I have only tested mapping downloader and not mapping uploader in this PR since my local machine does not allow uploading ~10GB of data into HDFS. 
 
if the attributes have not changed, I am pretty much sure the uploader script will work but we can test it on the spark cluster since we don't have any critical data in the cluster related to recommendations.

vansika commented on listenbrainz_spark/ftp/download.py:
```diff
@@ -94,21 +94,68 @@ def download_spark_dump_and_get_path(self, directory, dump_id, ftp_dump_dir, dum
+
+        mapping = list()
+        for mapping_name in dump:
+            mapping_pattern = '{}-\\d+-\\d+(.tar.bz2)$'.format(mapping_type)
```
@alastair I really think that the pattern can be improved? Can you have a look?

mayhem commented:
The columns have not changed -- I did fix the problem with that matchable field that wasn't matchable, but all else remains the same.

mayhem commented on listenbrainz_spark/ftp/download.py:
```diff
@@ -94,21 +94,68 @@ def download_spark_dump_and_get_path(self, directory, dump_id, ftp_dump_dir, dum
+                latest_mapping (str): latest mapping dump name.
+        """
+        # sort the mappings on timestamp
+        sorted_mapping = sorted(mapping, key=lambda x: int(x.split('-')[-2] + x.split('-')[-1].split('.')[0]))
```
This is a bit confusing -- you have a good regexp up above -- why not use that to parse the filename? Also, you should sort on the date string, not on the timestamp. The timestamp is only there to provide disambiguation in case we ever produce more than 1 dump in any given day.

vansika replied:
ummmm, so if we produce more than one dump in any given day, we would want to download the latest mapping of the two (of greater datetime)?

mayhem replied:
Yes

vansika replied:
So we should sort on date and timestamp both?

mayhem replied:
Yes

vansika replied:
Yeah. Was a bit confused by the original comment. Thanks :))

vansika commented on listenbrainz_spark/config.py.sample:
```diff
@@ -38,6 +38,11 @@ SENTRY_DSN_PUBLIC = ''
+# There are three types of mapping, namely 'msid-mbid-mapping', 'msid-mbid-mapping-with-text' and 'msid-mbid-mapping-with-matchable'
+# The type can be changed to download desired mapping
+# Note: Right now, we only have schema to support 'msid-mbid-mapping-with-matchable'
+MAPPING_TYPE = 'msid-mbid-mapping-with-matchable'
```
I am taking out this config option for now.

yvanzo commented:
Not sure if it applies here, but maybe the directory structure should implement [OTHER-17](https://tickets.metabrainz.org/browse/OTHER-17) as in <http://ftp.musicbrainz.org/pub/musicbrainz/data/json-dumps/>?

vansika commented on listenbrainz_spark/ftp/download.py:
```diff
@@ -94,21 +94,69 @@ def download_spark_dump_and_get_path(self, directory, dump_id, ftp_dump_dir, dum
+            _time = res[1]
+            return int(_date + _time)
+
+        sorted_mapping = sorted(mapping, key=callback)
```
umm... if there are no mappings available, an error will be raised by self.get_mapping_dump_name and this func would never be called. But can make the change. Looks good :)

mayhem replied:
ah, ok.

vansika commented on listenbrainz_spark/ftp/download.py:
```diff
@@ -94,21 +94,69 @@ def download_spark_dump_and_get_path(self, directory, dump_id, ftp_dump_dir, dum
 
-    def download_msid_mbid_mapping(self, directory, mapping_dump_id=None):
-        """ Download msid_mbid_mapping to dir passed as an argument.
+    def get_mapping_dump_name(self, dump, mapping_name_prefix):
```
cool :)

pep8speaks commented:
Hello @vansika! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`listenbrainz_spark/ftp/tests/test_download.py`](https://github.com/metabrainz/listenbrainz-server/blob/39603d7ad8826be1a7fccaf7c1b5e20523a2b358/listenbrainz_spark/ftp/tests/test_download.py):

> [Line 75:9](https://github.com/metabrainz/listenbrainz-server/blob/39603d7ad8826be1a7fccaf7c1b5e20523a2b358/listenbrainz_spark/ftp/tests/test_download.py#L75): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (2)

vansika commented:
`download_spark_dump_and_get_path` was earlier used for mapping as well as artist relations. Since we have changed the dir format of mappings, the function is not of much use. I have removed it and updated `download_artist_relation` accordingly. The PR now looks good to me.

mayhem commented:
@yvanzo : It does, but not directly to this project, since this project is not generating this data. Next time I work with the project that does generate this data, I'll all the LATEST file.

vansika commented:
@paramsingh ready for merge.
